### ISSUE TITLE
Euphonic: Powder averaging project

### ISSUE THREAD

**Issue by stfc-lam** [pace-neutrons/Pace-Project-Plan#25]

**Comment by ajjackson** [comment]
Compared to the semi-analytic method with incoherent and isotropic approximations, this refers to an implementation of powder-averaged **q** that avoids those approximations. Coherent scattering can be a significant or even dominant INS contribution depending on the material, so it is essential that some method is available.

The "obvious" method is spherical averaging in which contributions at regular scalar _q_ samples are averaged over uniform **q** samples from the surface of a sphere with radius _q_. In principle the necessary area can be reduced by symmetry, and convergence is systematic. In practice, the orientation of the samples could be quite important?

An alternative method is random sampling in which the spectrum is assembled from many random **q** vectors. This may avoid certain systematic biases and errors. It may also converge slowly and have difficulty capturing fine features.

Performance will be a challenge here. It would be useful to have a discussion about what exists and has already been tried with these methods. There might be ways of using geometric/band-structure information to inform effective sampling?

For now we are not considering the problem of multi-phonon terms as this leads to large numbers quite quickly. Some kind of scoping/test work would be nice to gauge how much error is associated with this simplification.

**Comment by ajjackson** [comment]
For now I assume that we are dealing with polycrystalline powders in which the coherent effects are observed within grains and boundary effects are not considered. A rough outline of work could be

## Stream 1
- [x] #85  Implement naive spherical averaging for coherent spectrum in crystalline samples (i.e. within a grain). This should take a force-constants model of a system and sample S at the surface of a sphere for a series of mod(q) values, with Debye-Waller contributions sampled from the whole Brillouin zone.
- [x] #86 (optional) From the above, discuss/determine feasibility of improving this by use of symmetry (and impact of angle-dependent instrument features)
- [x] #116 Explore convergence properties / adaptive sampling strategies for spherical averaging

## Stream 2
- [ ] #115 Explore approaches for spherical averaging from random samples
- [ ] #88  Implement a naive random-sampling implementation. This should sample 3D q-space for a fixed number of iterations, building up an S(mod(q), omega) spectrum.
- Explore the convergence properties of this method and propose strategies for a) efficient convergence of the spectrum b) confidence that the results have converged

## Then
- Compare results from spherical averaging and random sampling and decide if there is value in offering both methods
- Finalise implementation with a user-friendly API

**Comment by stfc-lam** [comment]
Refine and split into separate tasks.

**Comment by ajjackson** [comment]
Depends on #85 
Depends on #86 
Depends on #88

**Comment by cmarooney-stfc** [comment]
@ajjackson Hi Adam - not strictly for this ticket - using it as a convenient hook to ask a question. In the last sprint you have two active tasks
- powder averaging CLI in euphonic
- investigatge powder averaging performance 
which don't have a ticket number attached. If you could tell me what the task numbers are, that would be helpful to keep track
thanks, Chris

**Comment by ajjackson** [comment]
These tickets now exist and are linked from #116. As #116 has been marked as done in the project board, I'll close it, check as done here and add references to the emerging Issues.

**Comment by ajjackson** [comment]
From #116:

> Initial exploration done. Future actions (as discussed in meeting, notes coming soon to project docs) are to:
> 
> Profile powder-averaging script as there seem to be some performance issues #162
> - Examine jitter in r-axis #163
> - Implement initial API/scripts for a 3-parameter scheme #161
> - Investigate Sobol sampling #164
> - Use powder-averaging as a Euphonic-Brille interface test #86

**Comment by ajjackson** [comment]
At this stage it seems that Stream 2 is essentially parked and we are moving to actions that complete/deploy stream 1.

**Comment by rebeccafair** [comment]
Superseded by https://github.com/pace-neutrons/Pace-Project-Plan/issues/180?

**Comment by ajjackson** [comment]
Right. The stream 2 options remain vaguely interesting but we already have something that works quite well and they don't obviously offer a drastic (if any) improvement. So I think they can be parked indefinitely for now.
